src/modules/auto-state-on-scroll-bounds.js

```diff
@@ -0,0 +1,181 @@
+/**
+ *  @author Deux Huit Huit
+ *
+ *  Auto state on scroll bounds
+ 
+ 	Container: Element that scrolls and on which the state will be changed.
+ 				It has 4 states, 2 per axis.
+ 		<add class="js-scroll-bounds-ctn" />
+ 		<add data-scroll-bounds-threshold="0.1" />
+ 		
+ 		<add data-x-start-state-add-class="" />
+	 	<add data-x-start-state-rem-class="" />
+	 	<add data-x-end-state-add-class="" />
+	 	<add data-x-end-state-rem-class="" />
+	 	<add data-y-start-state-add-class="" />
+	 	<add data-y-start-state-rem-class="" />
+	 	<add data-y-end-state-add-class="" />
+	 	<add data-y-end-state-rem-class="" />
+ 	
+ 	Content: Element that is inside the container and that contains all of the content.
+ 			 Will be used to evaluate total scrolling width.
+ 		<add class="js-scroll-bounds" />
+ */
+(function ($, undefined) {
+	
+	'use strict';
+	var win = $(window);
+	var site = $('#site');
+	var scrollTimer = 0;
+	var resizeTimer = 0;
+	
+	var elements = $();
+	var defaultThreshold = 0;
+	
+	var SELECTOR_CTN = '.js-scroll-bounds-ctn';
+	var SELECTOR_SCROLL = '.js-scroll-bounds';
+	
+	var ATTR_THRESHOLD = 'data-scroll-bounds-threshold';
+	
+	var getBounds = function (elems) {
+		var b = {
+			x: 0,
+			y: 0
+		};
+		
+		elems.each(function () {
+			var t = $(this);
+			var x = t.outerWidth() + t.offset().left;
+			var y = t.outerHeight() + t.offset().top;
+			
+			b.x = Math.max(x, b.x);
+			
+			if (y > b.y) {
+				b.y = y;
+			}
+		});
+		
+		b.x = Math.floor(b.x);
+		b.y = Math.floor(b.y);
```

Comment:
```js
b.y = ~~b.y
```

is faster

Reply:
Yes, but I wonder if the clarity of the code should prevail in this case?

Reply:
Tor the case of the scroll event, performance win than clarity. Just add a comment for explaining why

Reply:
Done!